modules/general/key-vault/diagnostics.bicep

```diff
@@ -0,0 +1,51 @@
+param keyVaultName string
+param location string
+
+// TODO: Support shipping diagnostics to log analytics?
+
+@description('The storage account name to be used for key vault diagnostic settings')
+param diagnosticsStorageAccountName string = ''
+
+@description('When true, an existing storage account be used for diagnotics settings; When false, the storage account is created/updated')
+param useExistingStorageAccount bool = false
+
+@description('Sets the retention policy for diagnostics settings data, in days')
+param diagnosticsRetentionDays int = 30
+
+
+resource key_vault 'Microsoft.KeyVault/vaults@2021-06-01-preview' existing = {
+  name: keyVaultName
+}
+
+var _diagnosticsStorageAccountName = empty(diagnosticsStorageAccountName) ? keyVaultName : diagnosticsStorageAccountName
+
+
+resource existing_storage_account 'Microsoft.Storage/storageAccounts@2021-06-01' existing = if (useExistingStorageAccount) {
```

Comment:
It's probably best as a separate issue/change, but for completeness do you think we should support using an existing storage account in a different resource group?

Reply:
I've added a new issue https://github.com/endjin/Endjin.RecommendedPractices.Bicep/issues/45